Tests/tests/test_TextToSpeech.cpp

```diff
@@ -0,0 +1,173 @@
+    EXPECT_THAT(response, ::testing::ContainsRegex(_T("\"success\":true")));
+}
+
+TEST_F(TTSInitializedTest,Test_GetAPIVersion_Success) {
+    EXPECT_EQ(string(""), plugin->Initialize(&service));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("getapiversion"), _T(""), response));
+    EXPECT_THAT(response, ::testing::ContainsRegex(_T("\"version\":1")));
+    EXPECT_THAT(response, ::testing::ContainsRegex(_T("\"success\":true")));
+}
+
+TEST_F(TTSInitializedTest,Test_IsTTSEnabled_Success) {
+    EXPECT_EQ(string(""), plugin->Initialize(&service));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("isttsenabled"), _T(""), response));
+    EXPECT_EQ(response, _T("{\"isenabled\":false,\"TTS_Status\":0,\"success\":true}"));
+}
+
+TEST_F(TTSInitializedTest,Test_IsListVoicesEmpty_Success) {
+    EXPECT_EQ(string(""), plugin->Initialize(&service));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("listvoices"), _T("{\"language\":\"en-US\"}"), response));
+    EXPECT_EQ(response, _T("{\"voices\":[],\"TTS_Status\":0,\"success\":true}"));
+}
+
+TEST_F(TTSInitializedTest,Test_Speak_Success) {
+    EXPECT_EQ(string(""), plugin->Initialize(&service));
+    EXPECT_EQ(Core::ERROR_GENERAL, handler.Invoke(connection, _T("speak"), _T("{\"text\": \"speech_123\"}"), response));
+    EXPECT_EQ(response, _T(""));
+}
+
+TEST_F(TTSInitializedTest,Test_isSpeaking_Success) {
+    EXPECT_EQ(string(""), plugin->Initialize(&service));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("isspeaking"), _T("{\"speechid\": 1}"), response));
+    EXPECT_EQ(response, _T("{\"speaking\":false,\"TTS_Status\":0,\"success\":true}"));
+}
+
+TEST_F(TTSInitializedTest,Test_getSpeechState_Success) {
+    EXPECT_EQ(string(""), plugin->Initialize(&service));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("getspeechstate"), _T("{\"speechid\": 1}"), response));
+    EXPECT_EQ(response, _T("{\"speechstate\":3,\"TTS_Status\":0,\"success\":true}"));
+}
+
+TEST_F(TTSInitializedTest,Test_Cancel_Success) {
+    EXPECT_EQ(string(""), plugin->Initialize(&service));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("cancel"), _T("{\"speechid\": 1}"), response));
+    EXPECT_EQ(response, _T("{\"TTS_Status\":0,\"success\":true}"));
+}
+
+TEST_F(TTSInitializedTest,Test_SetTTSConfiguration_Success) {
+    EXPECT_EQ(string(""), plugin->Initialize(&service));
+    EXPECT_EQ(
+    Core::ERROR_NONE,
+    handler.Invoke(
+        connection,
+        _T("setttsconfiguration"),
+        _T("{\"language\": \"en-US\",\"voice\": \"carol\",\"ttsendpointsecured\":\"http://cdn-ec-ric-312.voice-guidance-tts.xcr.comcast.net/tts/v1/cdn/location?\",\"ttsendpointsecured\":\"http://cdn-ec-ric-312.voice-guidance-tts.xcr.comcast.net/tts/v1/cdn/location?\"}"),
+        response
+    )
+    );
+    EXPECT_EQ(response, _T("{\"TTS_Status\":0,\"success\":true}"));
+}
+
+TEST_F(TTSInitializedTest,Test_SetACL_Success) {
```

Comment:
Should we not test, if the SetACL is really preventing the calls from unauthorized clients?